cardano-testnet/test/cardano-testnet-test/Cardano/Testnet/Test/Gov/NoConfidence.hs

```diff
@@ -0,0 +1,261 @@
+                                        configurationFile
+                                        socketPath
+                                        (EpochNo 10)
+
+  governanceActionIndex <- case propSubmittedResult of
+                             Left e ->
+                               H.failMessage callStack
+                                 $ "findCondition failed with: " <> displayError e
+                             Right Nothing ->
+                               H.failMessage callStack "Couldn't find proposal."
+                             Right (Just a) -> return a
+
+  let spoVotes :: [(String, Int)]
+      spoVotes =  [("yes", 1), ("yes", 2), ("yes", 3)]
+      drepVotes :: [(String, Int)]
+      drepVotes = [("yes", 1), ("yes", 2), ("yes", 3)]
+
+  spoVoteFiles <- SPO.generateVoteFiles ceo execConfig work "spo-vote-files"
+                   governanceActionTxId governanceActionIndex
+                   [(defaultSPOKeys idx, vote) | (vote, idx) <- spoVotes]
+  drepVoteFiles <- DRep.generateVoteFiles execConfig work "drep-vote-files"
+                    governanceActionTxId governanceActionIndex
+                    [(defaultDRepKeyPair idx, vote) | (vote, idx) <- drepVotes]
+
+  let allVoteFiles = spoVoteFiles ++ drepVoteFiles
+  annotateShow allVoteFiles
+
+  -- Submit votes
+  voteTxBodyFp <- createVotingTxBody execConfig epochStateView sbe work "vote-tx-body"
+                                     allVoteFiles wallet0
+  let spoSigningKeys = [defaultSPOKeyPair n | (_, n) <- spoVotes]
+      drepSigningKeys = [defaultDRepKeyPair n | (_, n) <- drepVotes]
+      allVoteSigningKeys = spoSigningKeys ++ drepSigningKeys
+
+  voteTxFp <- signTx execConfig cEra work "signed-vote-tx" voteTxBodyFp
+                (paymentKeyInfoPair wallet0 : allVoteSigningKeys)
+
+  submitTx execConfig cEra voteTxFp
+
+  -- Step 4. We confirm the no confidence motion has been ratified by checking
+  -- for an empty constitutional committee.
+
+  mCommitteeEmpty
+    <- H.leftFailM $ findCondition (committeeIsPresent False) configurationFile socketPath (EpochNo 5)
+  H.nothingFail mCommitteeEmpty
+
+-- | Checks if the committee is empty or not.
+committeeIsPresent :: Bool -> AnyNewEpochState -> Maybe ()
+committeeIsPresent committeeExists (AnyNewEpochState sbe newEpochState) =
+  caseShelleyToBabbageOrConwayEraOnwards
+    (const $ error "Constitutional committee does not exist pre-Conway era")
+    (const $ let mCommittee = newEpochState
+                                ^. L.nesEsL
+                                 . L.esLStateL
+                                 . L.lsUTxOStateL
+                                 . L.utxosGovStateL
+                                 . L.cgsCommitteeL
+            in if committeeExists
+               then if isSJust mCommittee
+                    then Just () -- The committee is non empty and we terminate.
```

Comment:
That's not enough. The list of members can be empty here, `mCommittee` can be:
```haskell
SJust (Committee {committeeMembers = fromList [], committeeThreshold = 0 % 1})"
```
 You need to check for that as well.

Reply:
Not in the case of a `NoConfidence` governance action:

```
...
      NoConfidence _ ->
        st
          & ensCommitteeL .~ SNothing
          & ensPrevCommitteeL .~ SJust (GovPurposeId govActionId)
...
```

However, yes, if you manually remove all of the committee members.